### PR TITLE
[oneDNN] Fixes inconsistency in semantics of check for oneDNN opts in python and C++

### PR DESCRIPTION
**Pull request by ashiqimranintel**
After oneDNN was made default [ https://github.com/tensorflow/tensorflow/commit/0a63b9da132f2160411312af72de9e733e73dd65 ], the semantics of the check for oneDNN optimizations in python no longer matched the C++ checks. This fix will make sure python and c++ have the same semantics. The fix requires reshuffling the code and moving the IsMKLEnabled() implementation into onednn_env_vars.cc file.

This PR fixes following unit test failures

//tensorflow/python/client:timeline_test_cpu,
//tensorflow/c/eager:c_api_distributed_test_cpu,
//tensorflow/python/eager:context_test_cpu,
//tensorflow/python/framework:config_test_cpu,
//tensorflow/python/framework:config_test_tpu,
//tensorflow/python/framework:node_file_writer_test_cpu

### CODE REVIEW

File: tensorflow/core/util/util.cc [hunk 1]
```diff
@@ -22,7 +22,6 @@ limitations under the License.
 #include "tensorflow/core/framework/device_factory.h"
 #include "tensorflow/core/lib/gtl/inlined_vector.h"
 #include "tensorflow/core/lib/strings/strcat.h"
-#include "tensorflow/core/platform/cpu_info.h"
```
Comment (penpornk) [inline tensorflow/core/util/util.cc:25]: Please also remove the implementation of `IsMKLEnabled()` in this file.

File: tensorflow/core/util/BUILD [hunk 2]
```diff
@@ -484,7 +484,7 @@ cc_library(
 cc_library(
     name = "port",
     srcs = ["port.cc"],
-    hdrs = ["port.h"],
+    hdrs = ["port.h", "onednn_env_vars.h"],
```
Comment (penpornk) [inline tensorflow/core/util/BUILD:487]: I don't think we need to add "onednn_env_vars.h" here. This target should just depend on `:onednn_env_vars`.

File: tensorflow/core/util/util.h [hunk 3]
```diff
@@ -18,6 +18,7 @@ limitations under the License.
 
 #include "tensorflow/core/framework/tensor_shape.h"
 #include "tensorflow/core/lib/core/stringpiece.h"
+#include "tensorflow/core/util/onednn_env_vars.h"
 
 namespace tensorflow {
```
Comment (penpornk) [inline tensorflow/core/util/util.h:23]: Please remove the declaration of `IsMKLEnabled()` in this file (lines 60-61).

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/core/util/BUILD (hunk 2 above)
Comment (ashiqimranintel) [reply tensorflow/core/util/BUILD:487]: Thanks for your comment, without including "onednn_env_vars.h", it was not getting built successfully"

File: tensorflow/core/util/util.cc (hunk 1 above)
Comment (ashiqimranintel) [reply tensorflow/core/util/util.cc:25]: Thanks for your comment, removed it as per request

File: tensorflow/core/util/util.h (hunk 3 above)
Comment (ashiqimranintel) [reply tensorflow/core/util/util.h:23]: Thanks for your comment, removed it as per request

File: tensorflow/core/util/BUILD (hunk 2 above)
Comment (penpornk) [reply tensorflow/core/util/BUILD:487]: Could you please try adding `deps = [":onednn_env_vars"]` instead?

Comment (ashiqimranintel) [reply tensorflow/core/util/BUILD]: Thanks for your suggestion, moved it as a dependency

**Review by penpornk — APPROVED** [review]
Thank you for the fix!

**Comment by penpornk** [comment]
@ashiqimranintel Could you please help fix [Windows Bazel](https://source.cloud.google.com/results/invocations/6677f7ef-dfee-426a-8325-2ff4d757810e/log) failure (cyclic dependency)? Thank you!

**Comment by ashiqimranintel** [comment]
@penpornk, I pushed a fix for cyclic dependency

**Review by penpornk — APPROVED** [review]
Thank you for the fix!

**Review by penpornk — CHANGES_REQUESTED** [review]
Looks like we still have [cyclic dependency](https://source.cloud.google.com/results/invocations/f514a039-d43b-4359-b8b3-b918a4bc4510/log) from `//tensorflow/core/kernels:ops_util`

**Comment by ashiqimranintel** [comment]
> Looks like we still have [cyclic dependency](https://source.cloud.google.com/results/invocations/f514a039-d43b-4359-b8b3-b918a4bc4510/log) from `//tensorflow/core/kernels:ops_util`

I removed //tensorflow/core/kernels:ops_util, it should remove the cyclic dependency, Thank you

**Review by penpornk — APPROVED** [review]
Thank you! Let's try again. In the future, please only include dependencies that the target needs.

**Comment by ashiqimranintel** [comment]
> Thank you! Let's try again. In the future, please only include dependencies that the target needs.

Sure, I will make sure that happens.

**Comment by penpornk** [comment]
From [Android Demo App error](https://source.cloud.google.com/results/invocations/b0b7bd4a-5918-426c-9aee-0b11374428e1/log):
```
In file included from tensorflow/core/common_runtime/process_util.cc:31:
./tensorflow/core/util/util.h:21:10: fatal error: 'tensorflow/core/util/onednn_env_vars.h' file not found
#include "tensorflow/core/util/onednn_env_vars.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I believe this is because `onednn_env_vars.h` is not in the header list for Android. 
Cherry-picks should be minimal. Can we keep `IsMKLEnabled()` in `util.h` for now? We only need the changes in `port.cc` and `test_util.py` to fix the inconsistencies.

In the future, I think `IsMKLEnabled()` should just be moved to `port.cc` if possible. We shouldn't keep it with oneDNN-only header files because `IsMKLEnabled()` is callable from any builds.

**Comment by ashiqimranintel** [comment]
> "tensorflow/core/util/onednn_env_vars.h"

We don't want this cherry pick right now, we have a workaround for testing, we will have a cleaner fix it in the master, Thank you for checking this issue

**Comment by penpornk** [comment]
Sounds good. Thank you!
If the cherry-pick is still open by the time the cleaner fix is done, I'll cherry-pick it. :)

**Comment by penpornk** [comment]
Closing the PR for now. Please feel free to reopen or open a new PR.

**Pull request closed without merging**